src/validators/complex.rs

```diff
@@ -0,0 +1,69 @@
+use pyo3::exceptions::PyValueError;
+use pyo3::prelude::*;
+use pyo3::sync::GILOnceCell;
+use pyo3::types::{PyComplex, PyDict, PyString, PyType};
+
+use crate::errors::{ErrorTypeDefaults, ToErrorValue, ValError, ValResult};
+use crate::input::Input;
+
+use super::{BuildValidator, CombinedValidator, DefinitionsBuilder, ValidationState, Validator};
+
+static COMPLEX_TYPE: GILOnceCell<Py<PyType>> = GILOnceCell::new();
+
+pub fn get_complex_type(py: Python) -> &Bound<'_, PyType> {
+    COMPLEX_TYPE
+        .get_or_init(py, || py.get_type_bound::<PyComplex>().into())
+        .bind(py)
+}
+
+#[derive(Debug)]
+pub struct ComplexValidator {}
```

Comment:
I think we _probably_ need a strict mode flag here at the very least.

What about constraints like `Le`, or `MultipleOf`? They might get painful, so I'm ok to defer those to later as long as they fail gracefully.

Reply:
Do we want to support constraints for complex numbers?

I thought about constraints initially when I checked other validators to get some ideas of how to implement this complex validator, but I don't think constraints are going to be useful. For example, `Le` and relevant constraints don't really make sense as complex numbers don't have a default ordering. Users can apply additional filtering ad hoc based on the ordering they define, but I don't think we can add that kind of constraints to the validator.

Reply:
I am happy for us to not support constraints for now.